osu.Game/Overlays/Settings/Sections/Maintenance/MigrationSelectScreen.cs

```diff
@@ -32,8 +41,29 @@ protected override void OnSelection(DirectoryInfo directory)
 
             try
             {
-                if (target.GetDirectories().Length > 0 || target.GetFiles().Length > 0)
+                var directoryInfos = target.GetDirectories();
+                var fileInfos = target.GetFiles();
+
+                if (directoryInfos.Length > 0 || fileInfos.Length > 0)
+                {
+                    // Quick test for whether there's already an osu! install at the target path.
+                    if (fileInfos.Any(f => f.Name == @"client.realm"))
+                    {
+                        dialogOverlay.Push(new ConfirmDialog("The target directory already seems to have an osu! install. Use this data instead?", () =>
```

Comment:
```suggestion
                        dialogOverlay.Push(new ConfirmDialog("The target directory already seems to have an osu! install. Use that data instead?", () =>
```
I thought "this data" refers to data in the current directory when I first read it. "That" should avoid the confusion.